### PR TITLE
Stock Span Documentation

### LINKED ISSUE #1059: Stock Span Problem Documentation
<hr>

## Description 📜

Stock Span Problem Documentation using stack

<hr>

## Domain of Contribution 📊

- [x] C/CPP

<hr>

### PR DESCRIPTION
**Pull request by sinu-02**
<hr>

## Description 📜

I have provided a documentation of stock span problem using stack where the problem statement has been explained ,the algorithm followed to solve the problem and the code has been provided in cpp .
Fixes #1059  

<hr>

## Type of change 📝

- [x] Doc#umentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] C/#CPP

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

<hr>

### CODE REVIEW

**Comment by sinu-02** [comment]
@Suvraneel  I have generated new PR (this one #1904  for #1059(Stock Span Problem) ).So that I can merge from C_CPP Directory than main stream.Please review this one.

**Comment by Suvraneel** [comment]
@kashishahuja2002
Ye wala zara review kar dena

**Review by kashishahuja2002 — COMMENTED** [review]
As this problem statement is from leetcode, mention the reference website of the problem statement.

**Comment by sinu-02** [comment]
@kashishahuja2002 ok! can u pls tell me how should I edit my documentation?

**Comment by Suvraneel** [comment]
@sinu-02 same branch mein karne seh, it'll reflect in the PR

**Comment by sinu-02** [comment]
@Suvraneel  @kashishahuja2002  I have made the required changes please look on to the Documentation #1904

**Comment by kashishahuja2002** [comment]
Great. Looks good to me; ready for merge. @Suvraneel

**Comment by Suvraneel** [comment]
@sinu-02  Add relative link to  Data Structures/readme

**Comment by sinu-02** [comment]
@Suvraneel  How should I do that, do I have to create a readme in my forked repository ??

**Review by Suvraneel — CHANGES_REQUESTED** [review]
@sinu-02  
1.
This really doesn't belong in this place...
We've made changes to the directory structure... Kindly place this somewhere suitable :
either under Data Structure/Stack or under Competitive Prog folder

Review : @kashishahuja2002

File: C_CPP/Data Structures/Stock Span Problem/Description.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+1. We are given the price of the stock for the given number of consecutive days in the form of array.
+2. We will find the greatest integer to the left of the particular pointed index with each traversal.
+3. For that we will create a stack and a vector.
+4. For the leftmost price in the array we will push -1 in the vector.
+5. For other elements we will find the greatest element to the left of the pointed index.
+6. With every traversal we will keep on pushing the elment and it's index to the stack.
+7. If the top of stack is lesser than the current pointed element then it has to be popped else that element index has to be added to the index.
+8. By following all these procedures we will get the index of all the greatest elemnts to the left of the pointed index.
+9. The final result i.e **the span of days** will be given by subtracting the index of the particular price in the array and the element present on the same index in the vector.
+10.This the algorithm I foolowed for this problem.
```
Comment (Suvraneel) [inline C_CPP/Data Structures/Stock Span Problem/Description.md:18]:
```suggestion
10.This the algorithm I followed for this problem.
```

File: C_CPP/Data Structures/Stock Span Problem/Description.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        }
+        s.push({a[i],i});
+    }
+    for(int i=0;i<v.size();i++)
+    {
+        cout<<i-v[i]<<" ";
+    }
+
+    return 0;
+}
```
Comment (Suvraneel) [inline C_CPP/Data Structures/Stock Span Problem/Description.md:80]: Add new line at EOF

File: C_CPP/Data Structures/Stock Span Problem/Description.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+7. If the top of stack is lesser than the current pointed element then it has to be popped else that element index has to be added to the index.
+8. By following all these procedures we will get the index of all the greatest elemnts to the left of the pointed index.
+9. The final result i.e **the span of days** will be given by subtracting the index of the particular price in the array and the element present on the same index in the vector.
+10.This the algorithm I foolowed for this problem.
+
+### NOTE
+This problem is very famous and common during the placement procedure.
+For reference of whole statement of problem you can click on Question link:-
+[Question link](https://leetcode.com/problems/online-stock-span/)
+#  CODE
```
Comment (Suvraneel) [inline C_CPP/Data Structures/Stock Span Problem/Description.md:24]: Add Time Complexity analysis

File: C_CPP/Data Structures/Stock Span Problem/Stock Span Problem.cpp [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        s.push({a[i],i});
+    }
+    for(int i=0;i<v.size();i++)
+    {
+        cout<<i-v[i]<<" ";
+    }
+
+    return 0;
+}
+
```
Comment (Suvraneel) [inline C_CPP/Data Structures/Stock Span Problem/Stock Span Problem.cpp:50]: Add sample I/O

**Review by Suvraneel — CHANGES_REQUESTED** [review]
@sinu-02  
1.
This really doesn't belong in this place...
We've made changes to the directory structure... Kindly place this somewhere suitable :
either under Data Structure/Stack or under Competitive Prog folder

Review : @kashishahuja2002

**Review by kashishahuja2002 — COMMENTED** [review]
Yaa, this is more of a CP content.

**Comment by sinu-02** [comment]
@Suvraneel  I have made all the required changes please look on to it once.

**Review by Suvraneel — APPROVED** [review]

**Comment by Suvraneel** [comment]
@sinu-02  Observe the changes & do it from next time

Content-wise, good work ❤️

**Pull request merged**